src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -306,6 +300,9 @@ public CSharpOperationFactory(SemanticModel semanticModel)
                 case BoundKind.StackAllocArrayCreation:
                 case BoundKind.TypeExpression:
                 case BoundKind.TypeOrValueExpression:
+                case BoundKind.CollectionLiteralExpression:
+                case BoundKind.UnconvertedCollectionLiteralExpression:
+                case BoundKind.CollectionLiteralSpreadElement:
```

Comment:
Please implement `IBoundNodeWithIOperationChildren` on these nodes so children are correctly picked up.

I wouldn't expect to see `UnconvertedCollectionLiteralExpression` here. If one makes it to this stage, that means it made it out of initial binding without being handled and we missed a `BindToNaturalType` or error recovery call somewhere.

Reply:
It was not sufficient to implement `IBoundNodeWithIOperationChildren` from `BoundCollectionLiteralExpression` because the `BoundExpression` elements included an additional `Add()` wrapper for the collection initializer cases that `ControlFlowGraphBuilder` in particular was not expecting with the existing handling of `INoneOperation`. Instead, the child nodes are added explicitly, with the `Add()` wrapper skipped.